lib/schema-controller.js

```diff
@@ -72,13 +72,54 @@ class SchemaController {
     return this.schemaBucket.getSchemas()
   }
 
-  // Schema Controller compilers holder
+  // Called when compilers are recreated from builders, but the compiler
+  // was overridden via the legacy functions 'setValidatorCompiler' or
+  // 'setSerializerCompiler.'  It exists as a separate function only
+  // because c8 flags an inline arrow function as uncovered.
+  wrapAsBuilder (compiler) {
+    return compiler
+  }
+
   setValidatorCompiler (validatorCompiler) {
+    // Set up as if the fixed validator compiler had been provided
+    // by a custom 'options.compilersFactory.buildValidator' that
+    // always returns the same compiler object. This is required because:
+    //
+    // - setValidatorCompiler must immediately install a compiler to preserve
+    //   legacy behavior
+    // - setupValidator will recreate compilers from builders in some
+    //   circumstances, so we have to install this adapter to make it
+    //   behave the same if the legacy API is used
+    //
+    // The cloning of the compilersFactory object is necessary because
+    // we are aliasing the parent compilersFactory if none was provided
+    // to us (see constructor.)
+    this.compilersFactory = Object.assign(
+      {},
+      this.compilersFactory,
+      { buildValidator: this.wrapAsBuilder.bind(this, validatorCompiler) })
```

Comment:
Can we maybe delete `wrapAsBuilder` and set `/* istanbul ignore next */`

Reply:
@metcoder95 I'd rather not disable coverage, if possible.  Turns out it was my mistake, there really isn't a test that uses the new code path in the **serializer** case, so C8 was right.   I just need to add a test case then I think these can be arrow functions.  I had hidden this test problem inadvertently by sharing the named function *embarrassed*

Reply:
update: got rid of `wrapAsBuilder`, added tests, rebased and squashed.  test case under discussion is temporarily commented out